pandas/tests/test_index.py

```diff
@@ -603,6 +603,157 @@ def test_shift(self):
         shifted.name = 'shifted'
         self.assertEqual(shifted.name, shifted.shift(1, 'D').name)
 
+    def test_union_naming_behavior(self):
```

Comment:
instead of doing it this way, put methods in ``TestBase``, which then every sub-class will test (on its type). Look at how ``test_set_name_methods`` work.